blog/content/post/2022-12-16-trenchboot-aem-for-qubesos.md

```diff
@@ -0,0 +1,367 @@
+In order to install the packages one has to send the Xen and GRUB RPMs to the
+Dom0. Please not that moving any external files or data to Dom0 is potentially
+dangerous. Ensure that your environment is safe and the RPMs have the right
+checksums after copying them to Dom0. If you don't know how to copy files to
+Dom0, refer to the [Qubes OS documentation](https://www.qubes-os.org/doc/how-to-copy-from-dom0/#copying-to-dom0).
+
+1. Even before installing packages, it is required to enable the
+   `current-testing` repository to avoid the need to install additional
+   dependencies:
+
+    ```bash
+    sudo qubes-dom0-update --enablerepo=qubes-dom0-current-testing
+    ```
+
+2. If the RPMs are inside Dom0 install them with the following command (asuming
+   you downloaded whole package):
+
+    ```bash
+    sudo dnf update \
+        python3-xen-4.17.0-3.fc32.x86_64.rpm \
+        xen-4.17.0-3.fc32.x86_64.rpm \
+        xen-hypervisor-4.17.0-3.fc32.x86_64.rpm \
+        xen-libs-4.17.0-3.fc32.x86_64.rpm \
+        xen-licenses-4.17.0-3.fc32.x86_64.rpm \
+        xen-runtime-4.17.0-3.fc32.x86_64.rpm \
+        grub2-common-2.06-1.fc32.noarch.rpm \
+        grub2-pc-modules-2.06-1.fc32.noarch.rpm \
+        grub2-pc-2.06-1.fc32.x86_64.rpm \
+        grub2-tools-2.06-1.fc32.x86_64.rpm \
+        grub2-tools-extra-2.06-1.fc32.x86_64.rpm \
+        grub2-tools-minimal-2.06-1.fc32.x86_64.rpm
+    ```
+
+3. Invoke `sudo grub2-install /dev/sdX` where is the letter representing the
+   disk with `/boot` partition.
+3. Additionally you will have to download SINIT ACM and place it in `/boot`
+   partition/directory so that GRUB will be able to pick it up. Note it is only
+   necessary if your firmware/BIOS does not include/place SINTI ACM in the Intel
+   TXT region. You may obtain all SINIT ACMs as described
+   [here](https://github.com/QubesOS/qubes-antievilmaid/blob/7561a4d724b9b0df8ba48d8f2735d3754961f87b/README#L177).
+   Copy the SINTI ACM suitable for your platform to `/boot` directory.
+   In case of Dell OptiPlex it will be `SNB_IVB_SINIT_20190708_PW.bin`.
+4. Install Qubes AEM packages with the following command, because Qubes OS 4.2
+   lacks AEM packages:
+
+    ```bash
+    qubes-dom0-update --enablerepo=qubes-dom0-unstable --enablerepo=qubes-dom0-current-testing anti-evil-maid
+    ```
+
+5. Enter the SeaBIOS TPM menu (hotkey `t`) and there choose the clear TPM
+   option. Then activate and enable the TPM by selecting the appropriate
+   options.
+6. Follow steps in [setup TPM for AEM](https://github.com/QubesOS/qubes-antievilmaid/blob/7561a4d724b9b0df8ba48d8f2735d3754961f87b/README#L147).
+7. The anti-evil-maid script may not work with LUKS2 in its current state, so
+   make a fix according to this [Pull Request](https://github.com/QubesOS/qubes-antievilmaid/pull/41/files)
+   if needed.
+8. Now is possible to [setup Qubes OS AEM device](https://github.com/QubesOS/qubes-antievilmaid/blob/7561a4d724b9b0df8ba48d8f2735d3754961f87b/README#L202).
```

Comment:
```suggestion
8. Now it is possible to [setup Qubes OS AEM device](https://github.com/QubesOS/qubes-antievilmaid/blob/7561a4d724b9b0df8ba48d8f2735d3754961f87b/README#L202).
```

Reply:
I wonder if grammarly was used, it should catch those obvious checks. Also it doesn't make sense to waste time on typo fixes, which should be catched by automated spellchecker.

Reply:
Applied grammarly now and fixed it.